ValveResourceFormat/IO/MaterialExtract.cs

```diff
@@ -0,0 +1,405 @@
+            ["g_tNormalB"] = new[] { ("RGB", "TextureNormalB"), ("A", "TextureRoughnessB") },
+
+            ["g_tMask"] = new[] { ("R", "TextureMask") },
+        },
+
+        ["vr_eyeball"] = new()
+        {
+            ["g_tColor"] = new[] { ("RGB", "TextureColor"), ("A", "TextureReflectance") },
+            ["g_tIris"] = new[] { ("RGB", "IrisNormal"), ("A", "IrisRoughness") },
+            ["g_tNormal"] = new[] { ("GA", "TextureNormal") },
+
+            ["g_tIrisMask"] = new[] { ("L", "TextureIrisMask") },
+            ["g_tSelfIllumMask"] = new[] { ("L", "TextureSelfIllumMask") },
+        },
+    };
+
+    public static readonly Dictionary<string, string> CommonTextureSuffixes = new()
+    {
+        { "TextureColor", "_color" },
+        { "TextureNormal", "_normal" },
+        { "TextureRoughness", "_rough" },
+        { "TextureMetalness", "_metal" },
+        { "TextureAmbientOcclusion", "_ao" },
+        { "TextureReflectance", "_refl"}
+    };
+
+    private static bool IsDefaultTexture(string textureFileName)
+        => textureFileName.StartsWith("materials/default/", StringComparison.OrdinalIgnoreCase);
+
+    private readonly Material material;
+    private readonly ResourceEditInfo editInfo;
+    private readonly IFileLoader fileLoader;
+
+    public MaterialExtract(Material material, ResourceEditInfo editInfo, IFileLoader fileLoader)
+    {
+        this.material = material;
+        this.editInfo = editInfo;
+        this.fileLoader = fileLoader;
+    }
+
+    public MaterialExtract(Resource resource, IFileLoader fileLoader = null)
+        : this((Material)resource.DataBlock, resource.EditInfo, fileLoader)
+    {
+    }
+
+    public ContentFile ToContentFile()
+    {
+        var vmat = new ContentFile
+        {
+            Data = Encoding.UTF8.GetBytes(ToValveMaterial())
+        };
+
+        foreach (var (type, filePath) in material.TextureParams)
+        {
+            var texture = fileLoader?.LoadFile(filePath + "_c");
+            if (texture is null)
+            {
+                continue;
+            }
+
+            var texturesToUnpack = GetUnpackedTextures(type, filePath, material.IntParams, material.VectorParams);
+
+            var vtex = new TextureExtract(texture).ToMaterialMaps(texturesToUnpack);
+            vmat.ExternalRefsHandled.Add(vtex);
+
+            foreach (var imageSubFile in vtex.SubFiles)
+            {
+                if (IsDefaultTexture(imageSubFile.FileName))
+                {
+                    continue;
+                }
+
+                imageSubFile.FileName = Path.GetFileName(imageSubFile.FileName);
+                vmat.SubFiles.Add(imageSubFile);
+            }
+        }
+
+        return vmat;
+    }
+
+    public static string OutTextureName(string texturePath, string desiredSuffix = null)
+    {
+        texturePath = Path.ChangeExtension(texturePath, null); // strip .vtex_c
```

Comment:
Verify that you're stripping the right things?

Reply:
It should never fail with the current CompileMaterial implementation.

Reply:
*should*, but this will cause silent issues if it does change (or is somehow a given a different file).